haystack/schema.py

```diff
@@ -159,7 +149,8 @@ def to_dict(self, field_map={}) -> Dict:
                 if self.content_type == "table" and isinstance(self.content, pd.DataFrame):
                     v = [self.content.columns.tolist()] + self.content.values.tolist()
             k = k if k not in inv_field_map else inv_field_map[k]
-            _doc[k] = v
+            if k not in ["__pydantic_initialised__", "__initialised__"]:
```

Comment:
It might be easier and more future-proof to simply exclude all dunder keys here:

```suggestion
            if not k.startswith("__"):
```

I would also move it at the top of the loop for visibility reasons:

```python
for k, v in self.__dict__.items():

    # Exclude Pydantic internal fields from the conversion process
    if k.startswith("__"):
        continue

    ...
```

Reply:
I have not done this change because it would remove all private fields, not only pydantic ones. I was unsure if anyone using the codebase, 'till today, would need this metadata. 
Because 'till today, never no one claimed about the usage of private fields. If you say it's secure to remove all, and it won't affect I can remove it. Pydantic one I'm certain because they are just to control its internal flow. So, no important data for anyone.